masonry/src/passes/update.rs

```diff
@@ -145,3 +146,86 @@ pub(crate) fn run_update_pointer_pass(root: &mut RenderRoot, root_state: &mut Wi
     // Pass root widget state to synthetic state create at beginning of pass
     root_state.merge_up(root.widget_arena.get_state_mut(root.root.id()).item);
 }
+
+// ----------------
+
+fn recurse_on_children(
```

Comment:
This looks extremely similar to the function in `passes/compose.rs`. Please document the differences here

Reply:
It's the same. I'll remove the duplicate once #522 is merged.

Reply:
Done.